mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/documentdialog/OnDocumentUpdateListener.java

```diff
@@ -0,0 +1,14 @@
+package com.mifos.mifosxdroid.dialogfragments.documentdialog;
+
+import com.mifos.objects.noncore.Document;
+
+/**
+ * Created by Tarun on 14-08-17.
+ */
+
+public interface OnDocumentUpdateListener {
```

Comment:
Instead of creating two listeners why not create one listener called ```DocumentChangeListener```

It can have three methods

```created(Document document)```
```updated(Document document)```
```failed(String error, int actionType)```

```actionType``` is just an idea maybe for the method implementer to determine what did the user to do. 
After this create an Adapter so that you don't have to implement all three methods always you can choose which one you need.

Let me know if you need help with Adapter Pattern in Java.

Reply:
I got the "three methods and an identifier for the action" part. However, I'll need some guidance regarding
_After this create an Adapter so that you don't have to implement all three methods always you can choose which one you need._

Reply:
Check this out 

https://docs.oracle.com/javase/7/docs/api/java/awt/event/MouseAdapter.html

Reply:
I read it but didn't find it applicable to the particular case of ours. Just adding an `identifier` to distinguish the error worked well. Do let me know if it's not fine.